src/Parser/Warlock/Demonology/Modules/Features/DEUptimeDreadstalkers.js

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+
+import Analyzer from 'Parser/Core/Analyzer';
+
+import SPELLS from 'common/SPELLS';
+import SpellLink from 'common/SpellLink';
+
+import { formatPercentage } from 'common/format';
+import StatisticBox from 'Main/StatisticBox';
+import SpellIcon from 'common/SpellIcon';
+
+const CALL_DREADSTALKERS_DURATION = 12;
+const MILLISECONDS = 1000;
+
+class DEUptimeDreadstalkers extends Analyzer{
+  total_empowered_time = 0;
+  unempowered_dreadstalkers = [];
+  call_dread_casts = 0;
+
+  get uptime(){
+    return this.total_empowered_time / (this.call_dread_casts * CALL_DREADSTALKERS_DURATION * MILLISECONDS);
+  }
+
+  get suggestionThresholds(){
+    return {
+      actual: this.uptime,
+      isLessThan: {
+        minor: 0.95,
+        average: 0.90,
+        major: 0.80,
+      },
+      style: 'percentage',
+    };
+  }
+
+  on_byPlayer_cast(event){
+    const spellId = event.ability.guid;
+    if(spellId === SPELLS.CALL_DREADSTALKERS.id){
+      this.unempowered_dreadstalkers.push(event.timestamp);
+      this.call_dread_casts += 1;
+    } else if(spellId === SPELLS.DEMONIC_EMPOWERMENT.id){
+      this.unempowered_dreadstalkers.forEach((e) => {
```

Comment:
This would be cleaner as a reducer; https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce